lib/Parse/ParseExpr.cpp

```diff
@@ -2169,6 +2169,12 @@ DeclNameRef Parser::parseDeclNameRef(DeclNameLoc &loc,
     baseNameLoc = consumeIdentifier(
         &baseNameId, /*allowDollarIdentifier=*/true);
     baseName = baseNameId;
+    // Special case: Diagnose access to the underlying storage of a lazy
+    // variable.
+    if (baseName.getIdentifier().str().startswith("$__lazy_storage_$_") &&
```

Comment:
Might be worth hoisting this to a string constant, since it's now duplicated.

Reply:
Wouldn't a better fix be to hide this inside consumeIdentifier() handling of `allowDollarIdentifier`?